fvm/transactionInvocator.go

```diff
@@ -163,17 +163,23 @@ func (i *TransactionInvocator) Process(
 		txError = fmt.Errorf("transaction invocation failed: %w", err)
 	}
 
-	if txError == nil {
-		txError = i.checkAccountStorageLimit(vm, ctx, proc, sth, programs)
+	// try to deduct fees even if there is an error, bot do not hide the original error
+	feesError := i.deductTransactionFees(env, proc)
+	if feesError != nil {
+		if txError == nil {
+			txError = feesError
+		} else {
+			// combine the two errors so none of them gets lost.
+			txError = fmt.Errorf("%w and %v", txError, feesError)
```

Comment:
we could use https://github.com/hashicorp/go-multierror. thought not sure if we need to report txError at this point, user hasn't paid the fees so we should ignore the outcome of execution.

Reply:
> thought not sure if we need to report txError at this point, user hasn't paid the fees so we should ignore the outcome of execution.
That is a good point! This could be used to get information through errors. I will make it so only the fees error is used here.